python/cross_service/resilient_service/recommendation_service.py

```diff
@@ -0,0 +1,116 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+import json
+import logging
+
+import boto3
+from botocore.exceptions import ClientError
+
+log = logging.getLogger(__name__)
+
+
+class RecommendationServiceError(Exception):
+    def __init__(self, table_name, message):
+        self.table_name = table_name
+        self.message = message
+        super().__init__(self.message)
+
+
+# snippet-start:[python.example_code.workflow.ResilientService_RecommendationService]
+class RecommendationService:
+    """
+    Encapsulates a DynamoDB table to use as a service that recommends books, movies,
+    and songs.
+    """
+    def __init__(self, table_name, dynamodb_client):
+        """
+        :param table_name: The name of the DynamoDB recommendations table.
+        :param dynamodb_client: A Boto3 DynamoDB client.
+        """
+        self.table_name = table_name
+        self.dynamodb_client = dynamodb_client
+
+    @classmethod
+    def from_client(cls, table_name):
+        """
+        Creates this class from a Boto3 client.
+
+        :param table_name: The name of the DynamoDB recommendations table.
+        """
+        ddb_client = boto3.client('dynamodb')
+        return cls(table_name, ddb_client)
+
+    def create(self):
+        """
+        Creates a DynamoDB table to use a recommendation service. The table has a
+        hash key named 'MediaType' that defines the type of media recommended, such as
+        Book or Movie, and a range key named 'ItemId' that, combined with the MediaType,
+        forms a unique identifier for the recommended item.
+
+        :return: Data about the newly created table.
+        """
+        try:
+            response = self.dynamodb_client.create_table(
+                TableName=self.table_name,
+                AttributeDefinitions=[{
+                    'AttributeName': 'MediaType',
+                    'AttributeType': 'S'},{
+                    'AttributeName': 'ItemId',
+                    'AttributeType': 'N'}],
+                KeySchema=[{
+                    'AttributeName': 'MediaType',
+                    'KeyType': 'HASH'}, {
+                    'AttributeName': 'ItemId',
+                    'KeyType': 'RANGE'}],
+                ProvisionedThroughput={
+                    'ReadCapacityUnits': 5,
+                    'WriteCapacityUnits': 5},
+                SSESpecification={
```

Comment:
Did you need encryption for this particular use case? Or is this a best practice?

Reply:
You're right. When I read the doc, I thought I wanted True but on rereading, now I realize I want False (which doesn't mean "no encryption", it just means AWS will do it for me and not charge me for it).